### PR TITLE
RFC process clarification

### PR DESCRIPTION
**Pull request by akukanov**
Following an internal team discussion, this PR updates the RFC documentation to reflect that
- at any state proposals can be developed iteratively in several revisions;
- there is no single strictly defined flow, and each state can be skipped;
- "umbrella" documents might be simultaneously in a few states, as necessitated by the progress of sub-features.

### CODE REVIEW

File: rfcs/experimental/README.md [hunk 1]
```diff
@@ -5,19 +5,20 @@ released as experimental features in the library. An experimental
...
-demonstrated interest from the community,
-acceptance of the required specification changes, etc. 
+These conditions might include demonstrated performance improvements, demonstrated
+interest from the community, acceptance of the required specification changes, etc.
 
-For features that require specification changes, the document might
-include wording for those changes or a link to any PRs opened
-against the specification.
+A document here needs to be updated if the corresponding feature undergoes
+modifications while remaining experimental.
```
Comment (vossmjp) [inline rfcs/experimental/README.md:18]: Should you add that the document may also be updated as the status of the listed exit conditions change. For example, if performance improvements have been demonstrated.

File: rfcs/template.md [hunk 2]
```diff
@@ -24,6 +24,8 @@ Depending on the kind of the proposal, the description should cover:
 - New use cases supported by the extension.
 - The expected performance benefit for a modification, supported with the data, if available.
 - The API of extensions such as class definitions and function declarations.
+- Key technical design aspects, sufficient to understand how the functionality should work
```
Comment (vossmjp) [inline rfcs/template.md:27]: Does this section sound too strong now given the relaxing of the process. For example, maybe "Depending on the kind of proposal, the description may cover the following. And these sections may be iteratively clarified as the proposal matures." Or something like this,

File: rfcs/experimental/README.md (hunk 1 above)
Comment (akukanov) [reply rfcs/experimental/README.md]: Thanks; I added a sentence for this.

File: rfcs/template.md (hunk 2 above)
Comment (akukanov) [reply rfcs/template.md:28]: Updated in line with your suggestion.

**Review by vossmjp — APPROVED** [review]
Looks good to me

**Review by danhoeflinger — APPROVED** [review]
LGTM, I think this reflects the changes we wanted to make.

**Pull request merged**